src/lib.rs

```diff
@@ -144,6 +145,24 @@ macro_rules! schema {
             fn incomplete_write(&self) -> bool {
                 self.incomplete_write
             }
+
+            fn compact_log<P: AsRef<Path>>(&self, path: P) -> Result<(), $crate::errors::Error> {
+                $(let ($table_name, writer) = self.$table_name.begin_transaction()?;)*
+
+                let mut data = vec![];
+                $(
+                    for (key, val) in $table_name.get_all() {
+                        data.push(helper_log::$table_name::insert(key, val));
+                    }
+                )*
+
+                let mut schema_path = path.as_ref().to_path_buf();
+                schema_path.push(std::any::type_name::<$schema_name>().replace(':', "_"));
```

Comment:
If you do end up needing to recompute this sort of thing, I don't think you should copy and paste it. make it a function or whatever. Though I think you're more likely to just ask the currently writer where it's writing.